src/app/commands/cmd_layer_visibility.cpp

```diff
@@ -45,16 +45,31 @@ bool LayerVisibilityCommand::onEnabled(Context* context)
 bool LayerVisibilityCommand::onChecked(Context* context)
 {
   const ContextReader reader(context);
-  const Layer* layer = reader.layer();
-  return (layer && layer->isVisible());
+  const SelectedLayers& selLayers = reader.site()->selectedLayers();
+  bool visible = false;
+  for (auto layer : selLayers)
+  {
+    if(layer && layer->isVisible())
+      visible = true;
+  }
+  return visible;
 }
 
 void LayerVisibilityCommand::onExecute(Context* context)
 {
   ContextWriter writer(context);
-  Layer* layer = writer.layer();
+  const SelectedLayers& selLayers = writer.site()->selectedLayers();
```

Comment:
No problem. I should add a function to get the real set of selected layers and avoid this kind of problems in the future and simplify code (also this little code is duplicated in several places, the `writer.site()` should have a function to solve this for us already).